src/storage-client/src/controller.rs

```diff
@@ -292,11 +292,48 @@ pub trait StorageController: Debug {
     /// Get the current configuration, including parameters updated with `update_parameters`.
     fn config(&self) -> &StorageConfiguration;
 
-    /// Acquire an immutable reference to the collection state, should it exist.
-    fn collection(
+    /// Returns the `CollectionDescription` for the given `GlobalId`.
+    fn collection_description(
         &self,
         id: GlobalId,
-    ) -> Result<&CollectionState<Self::Timestamp>, StorageError<Self::Timestamp>>;
+    ) -> Result<CollectionDescription<Self::Timestamp>, StorageError<Self::Timestamp>>;
+
+    /// Returns the [CollectionMetadata] of the collection identified by `id`.
+    fn collection_metadata(
+        &self,
+        id: GlobalId,
+    ) -> Result<CollectionMetadata, StorageError<Self::Timestamp>>;
+
+    /// Returns the since/upper frontiers of the identified collections.
+    ///
+    /// Having a method that returns both frontiers at the same time, for all
+    /// requested collections, ensures that we can get a consistent "snapshot"
+    /// of collection state. If we had separate methods instead, and/or would
+    /// allow getting frontiers for collections one at a time, it could happen
+    /// that collection state changes concurrently, while information is
+    /// gathered.
```

Comment:
Is there any part of the code that needs that snapshot guarantee?

Reply:
Not right now, but also felt more correct. But I'm not invested in this.